src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/BaseAnalysisServiceImpl.java

```diff
@@ -363,13 +365,85 @@ public Boolean moveAnalysis(Long id, Integer index) {
         return true;
     }
 
+    protected boolean detectExecutable(CommonAnalysisType type, MultipartFile file) {
+
+        return false;
+    }
+
+    @Override
+    public List<AnalysisFile> saveFiles(List<UploadFileDTO> files, IUser user, A analysis, List<String> errorFileMessages) {
+
+        files.forEach(f -> {
```

Comment:
In fact validation logic has completely moved to service layer, 
from my point of view that is not a good idea.
We can discuss more details in offline.